Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.SwitchExpression.cs

```diff
@@ -49,18 +50,22 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public SwitchExpression Update (Expression switchValue, IEnumerable<SwitchCase> cases, Expression defaultBody)
     {
+      Contract.Requires(switchValue != null);
```

Comment:
And again, this is not the case: http://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/SwitchExpression.cs,204

Any enumerable that is used in a form of `someEnumerable.ToReadOnly()` could be null.